src/lib/y2storage/proposal/assigned_space.rb

```diff
@@ -56,12 +56,14 @@ def partition_type
         @partition_type if @partition_type_calculated
 
         @partition_type_calculated = true
-        @partition_type = if disk.partition_table.extended_possible
-          if disk.partition_table.has_extended
-            inside_extended? ? :logical : :primary
+        disk.as_not_empty do
```

Comment:
Actually, there is no different branch for completely empty disks. We are still doing the same. But now instead of raising an exception if the partition table is not there we assume that, even if it's not there now, there would be a GPT partition as soon as it's needed.

That's the general idea of using `as_not_empty`. It dresses up completely empty disks as disks with an empty GPT table, so they are considered like so during the calculations (free spaces, logical vs primary, etc.)